sdk/eventgrid/Azure.Messaging.EventGrid/README.md

```diff
@@ -197,24 +197,37 @@ Below is an example calling `GetData()` using Event Grid events. Calling `GetDat
 ```C# Snippet:DeserializePayloadUsingNonGenericGetData
 foreach (EventGridEvent egEvent in egEvents)
 {
-    // If the event is a system event, GetData() should return the correct system event type
-    switch (egEvent.GetData())
+    // If the event is a system event, AsSystemEventData() should return the correct system event type
+    if (egEvent.IsSystemEvent)
     {
-        case SubscriptionValidationEventData subscriptionValidated:
-            Console.WriteLine(subscriptionValidated.ValidationCode);
-            break;
-        case StorageBlobCreatedEventData blobCreated:
-            Console.WriteLine(blobCreated.BlobType);
-            break;
-        case BinaryData unknownType:
-            // An unrecognized event type - GetData() returns BinaryData with the serialized JSON payload
-            if (egEvent.EventType == "MyApp.Models.CustomEventType")
-            {
+        switch (egEvent.AsSystemEventData())
```

Comment:
No, it wouldn't.

Reply:
This is one of my questions as well - if it would make sense to throw an exception from AsSystemEventData() instead of returning null if the event is not a system event type.  We do this in FormRecognizer where we have AsXx() methods on [FieldValue](https://apiview.dev/Assemblies/Review/cf2c57b3d4c44ef08c2e8284f7441a2d#Azure.AI.FormRecognizer.Models.FieldValue).  If users will always use this in a switch statement, maybe it doesn't matter, but it might be worth thinking about whether  you'd ever add TryGetAs methods for this in the future, would the returning null semantics make sense?

Reply:
I don't think throwing an exception would be ideal since someone who only works with system events might get into trouble if a topic they are subscribing to starts getting custom events (and their app doesn't care about these custom events). Instead of just gracefully skipping these, their app would start throwing exceptions.

Reply:
I don't think we would add TryGetAs in the future, so we should choose one or the other for GA and stick with it.